spyder_kernels/customize/spydercustomize.py

```diff
@@ -783,13 +818,25 @@ def _get_globals():
     return ipython_shell.user_ns
 
 
+def _frontend_request(blocking=True):
+    if not get_ipython().kernel.frontend_comm.is_open():
+        raise CommError("Can't make a request to a closed comm")
+    return get_ipython().kernel.frontend_call(blocking=blocking)
+
+
 def runfile(filename, args=None, wdir=None, namespace=None, post_mortem=False):
     """
     Run filename
     args: command line arguments (string)
     wdir: working directory
     post_mortem: boolean, whether to enter post-mortem mode on error
     """
+    try:
+        # Save the open files
+        _frontend_request().savefiles()
+    except (CommError, TimeoutError):
+        logger.debug("Could not send save files before executing.")
```

Comment:
I don't think this should be part of this PR. It's new functionality, so it should go in a different PR.